File: turtle_circle.cpp
#include<rclcpp/rclcpp.hpp>
#include"geometry_msgs/msg/twist.hpp"
#include"chrono"

   class TurtleCircleNode: public rclcpp::Node
   {
   private:
      rclcpp::TimerBase::SharedPtr timer_;
      rclcpp::Publicsher<geometry_msgs::msg::Twist>::SharedPtr Publicsher_;  //发布者的智能指针
   public:
      explicit TurtleCircleNode(const std::string& node_name):Node(node_name)
      {
         publicher_ = this->create_publisher<geometry_msgs::msg::Twist>("turtle1_cmd_vel", 10);
         timer_ = this->create_wall_timer(1000ms,std::bind(&TurtleCircleNode::timer_callback,this));
      }
      void timer_callback()
      {
         auto msg = geometry_msgs::msg::Twist();
         msg.linear.x = 1.0;
         msg.linear.z = 0.5;
         publisher_->publish(msg);
      }

   } ;

   int main(int argc,char* argv[])
   {
      rclcpp::init(argc,argv);
      auto node = std::make_shared<TurtleCircleNode>("turtle_node");
      rclcpp::spin(node);
      rclcpp::shutdown();
   }

File: turtle_circle.py
import rclpy
from rclpy import Node
from geometry_msgs.msg import Twist

class TurtleCircleNode(Node):
    def __init__(self, node_name):
        super().__init__(node_name)
        self.get_logger().info(f'{node_name}节点，启动！')
        self.TurtleCircle_publisher = self.create_publisher(Twist,'/turtle1/cmd_vel',10)  # 创建话题
        self.create_timer(1,self.timer_callback)  # 定义计时器 1秒一次

    def timer_callback(self):
        twist = Twist()  # 组装数据
        twist.linear.x = 1.0
        twist.angular.z = 0.5
        self.TurtleCircle_publisher.publish(twist)  # 发布话题
        self.get_logger().info(f'发布了话题{twist}')


def main():
    rclpy.init()
    node = TurtleCircleNode('TurtleCircleNode')
    rclpy.spin(node)
    rclpy.shutdown()


